lib/code_corps/stripe_service/stripe_connect_account.ex

```diff
@@ -16,43 +14,17 @@ defmodule CodeCorps.StripeService.StripeConnectAccountService do
     end
   end
 
-  def add_external_account(%StripeConnectAccount{id_from_stripe: stripe_id} = record, external_account) do
-    with {:ok, %Stripe.Account{} = stripe_account} <- @api.Account.update(stripe_id, %{external_account: external_account}),
-         {:ok, params} <- StripeConnectAccountAdapter.to_params(stripe_account, %{})
-    do
-      record
-      |> StripeConnectAccount.webhook_update_changeset(params)
-      |> Repo.update
-    end
-  end
-
-  def add_vertification_document(
-    %StripeConnectAccount{id_from_stripe: account_id} = record,
-    %{"legal_entity_verification_document" => document} = attributes
-  ) do
-    with {:ok, %Stripe.FileUpload{} = stripe_file_upload} <- Stripe.FileUpload.retrieve(document),
-         {:ok, %Stripe.Account{} = stripe_account} <- Stripe.Account.update(account_id, %{legal_entity: %{verification: %{document: document}}}),
-         {:ok, file_upload_params} <- StripeFileUploadAdapter.to_params(stripe_file_upload, attributes),
-         {:ok, connect_account_params} <- StripeConnectAccountAdapter.to_params(stripe_account, attributes),
-         account_changeset <- record |> StripeConnectAccount.webhook_update_changeset(connect_account_params),
-         file_changeset <- %StripeFileUpload{} |> StripeFileUpload.create_changeset(file_upload_params)
+  def update(%StripeConnectAccount{id_from_stripe: id_from_stripe} = account, %{} = attributes) do
+    with {:ok, from_params} <- StripeConnectAccountAdapter.from_params(attributes),
+         {:ok, %Stripe.Account{} = stripe_account} <- @api.Account.update(id_from_stripe, from_params),
+         {:ok, params} <- StripeConnectAccountAdapter.to_params(stripe_account, attributes),
+         {:ok, %StripeConnectAccount{} = updated_account} <- account |> StripeConnectAccount.webhook_update_changeset(params) |> Repo.update
     do
-      multi =
-        Multi.new
-        |> Multi.update(:stripe_connect_account, account_changeset)
-        |> Multi.insert(:stripe_file_upload, file_changeset)
-
-      case Repo.transaction(multi) do
-        {:ok, %{stripe_connect_account: account, stripe_file_upload: _}} ->
-          {:ok, account}
-        {:error, :stripe_connect_account, %Ecto.Changeset{} = account_changeset, %{}} ->
-          {:error, account_changeset}
-        # If creating the file failed due to validation, we add a generic error
-        # to the account changeset and return that to be rendered.
-        {:error, :stripe_file_upload, %Ecto.Changeset{} = _, _} ->
-          account_changeset |> Ecto.Changeset.add_error(:legal_entity_verification_document, "is invalid")
-          {:error, account_changeset}
-      end
+      {:ok, updated_account}
+    else
+      {:error, %Ecto.Changeset{} = changeset} -> {:error, changeset}
+      {:error, %Stripe.APIErrorResponse{} = error} -> {:error, error}
+      _ -> {:error, :unhandled}
```

Comment:
We will likely want to sweep through the code at some point and remove these `_ -> unhandled` cases. They basically swallow any unexpected errors in the process. We should explicitly list any failures we support and handle. If anything unexpected fails, we should treat it as a bug. 

This is basically equivalent with us wrapping a huge chunk of code into a `try-catch` that catches any sort of exception.